tools/tf2ngraph.py

```diff
@@ -334,6 +335,83 @@ def convert(inp_format, inp_loc, out_format, out_loc, output_nodes, ng_backend,
+    '''
+    # get rid of caret indicating control edge (^name -> name)
+    if node_name.startswith('^'):
+        node_name = node_name[1:]
+
+    # get rid of output slot (name:0 -> name)
+    split_colon = node_name.split(':')
+    if len(split_colon) == 1 or len(split_colon) == 2:
+        return split_colon[0]
+    else:
+        raise NotImplemented(
+            "Expected name to have <= 1 colons. Handle case with > 1 colons")
+
+
+def guess_output_nodes(graph_def):
+    '''
+    Given a graph def guess the outputs and ask user to select one from the candidates
+    '''
+    nodes_which_appear_at_inputs = set()
+    for n in graph_def.node:
+        # the list comprehension converts a
+        # google.protobuf.pyext._message.RepeatedScalarContainer to a list of strings
+        nodes_which_appear_at_inputs.update(
+            [sanitize_node_name(i) for i in n.input])
+
+    all_node_names = {n.name for n in graph_def.node}
+    possible_outputs = all_node_names.difference(nodes_which_appear_at_inputs)
+    name_type_map = get_name_type_map(graph_def)
+    return {k: name_type_map[k] for k in possible_outputs}
+
+
+def get_name_type_map(graph_def):
+    return {n.name: n.op for n in graph_def.node}
+
+
+def infer_output_nodes(inp_format, inp_loc):
+    '''
+    Try to read output names from savedmodel's signature_def
+    or try to guess the outputs from the graphdef
+    '''
+    if inp_format == 'savedmodel':
+        with tf.Session(graph=tf.Graph()) as sess:
+            # load the saved model
+            imported = tf.saved_model.load(
+                sess,
+                tags=[tf.saved_model.tag_constants.SERVING],
+                export_dir=inp_loc)
+            try:
+                # Check if the saved model has outputs specified
+                output_info = imported.signature_def[
+                    'serving_default'].outputs.values()
+                saved_model_has_out_name = True
+            except:
+                saved_model_has_out_name = False
+            if saved_model_has_out_name:
+                # the list comprehension gets the names
+                # from the output_info of type collections.abc.ValuesView
+                possible_outputs = [
+                    sanitize_node_name(i.name) for i in output_info
+                ]
+                node_name_type_dict = get_name_type_map(imported.graph_def)
+                return {k: node_name_type_dict[k] for k in possible_outputs}
+            else:
+                # Outputs not specified in the saved model
+                # Hence using guess_output_nodes
+                return guess_output_nodes(sess.graph_def)
+    elif inp_format == 'pbtxt' or inp_format == 'pb':
+        return guess_output_nodes(get_gdef_from_protobuf(inp_loc))
+    else:
+        raise ValueError(
+            "inp_format expected to be pb, pbtxt or savedmodel, but found ",
+            inp_format)
+
+
 def main():
```

Comment:
Unfriendly error message: Please fix this:
```
Traceback (most recent call last):
  File "../ngraph-bridge/tools/tf2ngraph.py", line 443, in <module>
    main()
  File "../ngraph-bridge/tools/tf2ngraph.py", line 423, in main
    out_format, out_loc = filter_dict("output", args.__dict__)
  File "../ngraph-bridge/tools/tf2ngraph.py", line 262, in filter_dict
    len(current_format)) + " " + prefix + " formats, expected only 1"
AssertionError: Got 0 output formats, expected only 1
```
Change this to indicate that there is missing output file format. Then display possible formats.

Reply:
Changed `Got 0 output formats, expected only 1` to `Got 0 output formats, expected only 1. Please add --output_pb or --output_pbtxt or --output_savedmodel and pass the output model location`